go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -188,29 +185,27 @@ func (ovn *Controller) createLoadBalancerRejectACL(lb, sourceIP string, sourcePo
 		ovn.removeACLFromNodeSwitches(switches, aclUUID)
 		return aclUUID, nil
 	}
-
-	aclMatch = fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=from-lport", "priority=1000", aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName)}
+	errList := []error{}
 	if applyToPortGroup {
-		cmd = append(cmd, "--", "add", "port_group", ovn.clusterPortGroupUUID, "acls", "@reject-acl")
+		aclUUID, err = acl.AddRejectACLToPortGroup(ovn.clusterPortGroupUUID, aclName, sourceIP, strings.ToLower(string(proto)), int(sourcePort))
+		if err != nil {
+			klog.Errorf("Failed to add LB: %v", err)
+			errList = append(errList, err)
+		}
 	}
 	if len(gwRouterExtSwitch) > 0 {
```

Comment:
@trozet I don´ t understand this part, `if applyToPortGroup {` and ` if len(gwRouterExtSwitch) > 0 `  are exclusive? should be a if.. else ? otherwise the later can override the former

Reply:
They are separate: applyToPortGroup is for applying to node switches, while gwRouterExtSwitch is for applying to an external switch. One is not a superset of the other.

Reply:
ok, then let me try to see if I get it correctly, a NodePort Service have 2 load balancers:
- one for the ClusterIP that has the ACLs applied the portgroup 
- and other in the NodePort Loadbalancer, that has the ACL applied in the GR switch

Reply:
almost:

1. one for the ClusterIP that has the ACLs applied the portgroup
2. and other in the NodePort Loadbalancer, that has the ACL applied in the GR switch and the portgroup

the node port load balancer exists on the GR and the node worker switch. This is so that we can handle a pod trying to access nodeport to go straight to its backend.

Reply:
> the node port load balancer exists on the GR and the node worker switch. This is so that we can handle a pod trying to access nodeport to go straight to its backend.

ahhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh this connect the dots ... thanks

Reply:
It works for me using only portgroups, NodePorts works too, without configuring the ACLs on the switches  ... am I missing something or testing it wrong ? why does it work?